File: models/Muscle.py
import json
class Muscle:
    def __init__(self, id, name, category, description):
        self.id = id
        self.name = name
        self.category = category
        self.description = description
      
    @staticmethod  
    def get_All_Muscle_by_AreaId(area_id):
        pass


    def get_Muscle_data(self):
        muscle_data = {
            "id": self.id,
            "name": self.name,
            "category": self.category,
            "description": self.description
        }
        return json.dumps(muscle_data)

File: view/lib/models/Muscle.dart
import 'dart:convert';

class Muscle {
  String id;
  String name;
  String describe;
  String sourReason;

  Muscle({
    required this.id,
    required this.name,
    required this.describe,
    required this.sourReason,
  });

  // 靜態方法來模擬從區域ID獲取所有肌肉
  static Future<List<Muscle>> getAllMuscleByAreaId(String areaId) async {
    // 實現你的邏輯，例如從API或數據庫獲取數據
    // 這裡我們返回一個空列表作為佔位符
    return [];
  }

  // 獲取肌肉數據
  String getMuscleData() {
    Map<String, dynamic> muscleData = {
      "id": id,
      "name": name,
      "describe": describe,
      "sour_reason": sourReason
    };
    return jsonEncode(muscleData);
  }
}
